addons/addon-base-raas/packages/base-raas-services/lib/alb/__tests__/alb-service.test.js

```diff
@@ -0,0 +1,433 @@
+        return null;
+      });
+      // OPERATE
+      const count = await service.albDependentWorkspacesCount({}, 'project-id');
+      // CHECK
+      expect(count).toEqual(0);
+    });
+  });
+
+  describe('check alb exists function', () => {
+    it('should fail because project id is not valid', async () => {
+      projectService.mustFind.mockImplementationOnce(() => {
+        throw service.boom.notFound(`project with id "test-id" does not exist`, true);
+      });
+      // OPERATE
+      try {
+        await service.checkAlbExists({}, 'test-id');
+      } catch (err) {
+        // CHECK
+        expect(err.message).toEqual('project with id "test-id" does not exist');
+      }
+    });
+
+    it('should return a true if alb arn present in deployment item', async () => {
+      service.getAlbDetails = jest.fn(() => {
+        return albDetails;
+      });
+      // OPERATE
+      const albExists = await service.checkAlbExists({}, 'project-id');
+      // CHECK
+      expect(albExists).toEqual(true);
+    });
+
+    it('should return false if deployment item is not found', async () => {
+      service.getAlbDetails = jest.fn(() => {
+        return null;
+      });
+      // OPERATE
+      const albExists = await service.checkAlbExists({}, 'project-id');
+      // CHECK
+      expect(albExists).toEqual(false);
+    });
+
+    it('should return false if alb arn is null found in deployment item', async () => {
+      const albDetailsNotExists = {
+        createdAt: '2021-05-21T13:06:58.216Z',
+        id: 'test-id',
+        type: 'account-workspace-details',
+        updatedAt: '2021-05-31T13:32:15.503Z',
+        value:
+          '{"id":"test-id","albStackName":null,"albArn":null,"listenerArn":null,"albDnsName":null,"albDependentWorkspacesCount":1}',
+      };
+      service.getAlbDetails = jest.fn(() => {
+        return albDetailsNotExists;
+      });
+      // OPERATE
+      const albExists = await service.checkAlbExists({}, 'project-id');
+      // CHECK
+      expect(albExists).toEqual(false);
+    });
+  });
+
+  describe('increase alb dependent workspace count function', () => {
+    it('should fail because project id is not valid', async () => {
+      projectService.mustFind.mockImplementationOnce(() => {
+        throw service.boom.notFound(`project with id "test-id" does not exist`, true);
+      });
+      // OPERATE
+      try {
+        await service.increaseAlbDependentWorkspaceCount({}, 'test-id');
+      } catch (err) {
+        // CHECK
+        expect(err.message).toEqual('project with id "test-id" does not exist');
+      }
+    });
+
+    it('should pass if the input is valid', async () => {
+      service.getAlbDetails = jest.fn(() => {
+        return albDetails;
+      });
+      service.saveAlbDetails = jest.fn(() => {
+        return { id: 'id-alb' };
+      });
+      service.audit = jest.fn();
+      // OPERATE
+      await service.increaseAlbDependentWorkspaceCount({}, 'test-id');
```

Comment:
Shouldn't this verify if the count was increased successfully or not?

Reply:
Made the change and updated the PR